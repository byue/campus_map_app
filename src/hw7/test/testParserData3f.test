# test parser on a graph with 3 linear nodes doubly connected

LoadGraph heroes data3f.tsv

ListChildren heroes a
ListChildren heroes b
ListChildren heroes c

FindPath heroes a b

FindPath heroes b c

FindPath heroes a c

AddEdge heroes a c 0.01

ListChildren heroes a

FindPath heroes a c

FindPath heroes a a