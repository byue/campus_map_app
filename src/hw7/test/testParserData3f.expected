# test parser on a graph with 3 linear nodes doubly connected

loaded graph heroes

the children of a in heroes are: b(0.500)
the children of b in heroes are: a(0.500) c(0.500)
the children of c in heroes are: b(0.500)

path from a to b:
a to b with weight 0.500
total cost: 0.500

path from b to c:
b to c with weight 0.500
total cost: 0.500

path from a to c:
a to b with weight 0.500
b to c with weight 0.500
total cost: 1.000

added edge 0.010 from a to c in heroes

the children of a in heroes are: b(0.500) c(0.010)

path from a to c:
a to c with weight 0.010
total cost: 0.010

path from a to a:
total cost: 0.000